File: src/main/java/de/steklopod/sec08/Lec04Zip.java
package de.steklopod.sec08;

import de.steklopod.courseutil.Util;
import reactor.core.publisher.Flux;

public class Lec04Zip {

    public static void main(String[] args) {
        Flux.zip(getBody(), getEngine(), getTires())
                .subscribe(Util.subscriber());

    }

    private static Flux<String> getBody(){
        return Flux.range(1, 5)
                .map(i -> "body");
    }

    private static Flux<String> getEngine(){
        return Flux.range(1, 3)
                .map(i -> "engine");
    }

    private static Flux<String> getTires(){
        return Flux.range(1, 6)
                .map(i -> "tires");
    }


}

File: src/main/kotlin/de/steklopod/sec08/Lec04Zip.kt
package de.steklopod.sec08

import de.steklopod.utils.Util.subscriber
import reactor.core.publisher.Flux

object Lec04Zip {
    @JvmStatic
    fun main(args: Array<String>) {
        Flux.zip(body, engine, tires)
            .subscribe(subscriber())
    }

    private val body: Flux<String>
        private get() = Flux.range(1, 5)
            .map { i: Int? -> "body" }
    private val engine: Flux<String>
        private get() = Flux.range(1, 3)
            .map { i: Int? -> "engine" }
    private val tires: Flux<String>
        private get() = Flux.range(1, 6)
            .map { i: Int? -> "tires" }
}
